### PR TITLE
docs: clarify JSON terms in COREDUMP_PACKAGE_METADATA

### PR DESCRIPTION
**Pull request by bluca**
As suggested on: https://sourceware.org/pipermail/elfutils-devel/2021q4/004402.html

### CODE REVIEW

File: docs/COREDUMP_PACKAGE_METADATA.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ the original executable, debuginfo and sources, to further facilitate debugging.
 SECTION: `.note.package`
```
Comment (fweimer-rh) [inline docs/COREDUMP_PACKAGE_METADATA.md:49]: While you are at it, maybe you can add that the section name is not significant?

Comment (bluca) [reply docs/COREDUMP_PACKAGE_METADATA.md:49]: sorry, what do you mean by that?

Comment (fweimer-rh) [reply docs/COREDUMP_PACKAGE_METADATA.md:49]: The name does not matter, consumers need to look at all note sections and parse the notes contained in them.

Comment (bluca) [reply docs/COREDUMP_PACKAGE_METADATA.md:49]: But it matters for a human reader, no? Someone running `readelf --notes` should see `.note.package`, and not `.note.foobar.baz`. For tools there's the ID of course.

Comment (bluca) [reply docs/COREDUMP_PACKAGE_METADATA.md:49]: I've added a line saying that when implementing parsers, the note should be searched using the id

Comment (bluca) [reply docs/COREDUMP_PACKAGE_METADATA.md:49]: @fweimer-rh does the following address your concern?

`and should look for the note using the node-id.`

Comment (fweimer-rh) [reply docs/COREDUMP_PACKAGE_METADATA.md:49]: *node-id* should probably be *note type*? Also in the section description. But otherwise, works for me.

Comment (bluca) [reply docs/COREDUMP_PACKAGE_METADATA.md:49]: thanks, fixed that too

**Pull request merged**